src/main/java/com/celements/web/plugin/cmd/CssCommand.java

```diff
@@ -205,42 +205,33 @@ public List<CSS> includeCSSAfterPageType(String css, XWikiContext context) {
     if ((pageTypeDoc != null) && (vcontext != null)) {
       List<BaseObject> baseList = new ArrayList<>();
       baseList.addAll(addUserSkinCss(pageTypeDoc));
-      baseList.addAll(addUserSkinCss((Document) vcontext.get("after_pagetype_cssdoc")));
       cssList = includeCSS(css, "cel_css_list_pagetype", baseList, context);
     }
     return cssList;
   }
 
   public List<CSS> includeCSSAfterPageLayout(String css, XWikiContext context) {
-    XWikiDocument pageLayoutDoc = null;
-    pageLayoutDoc = new PageLayoutCommand().getLayoutPropDoc();
+    Optional<DocumentReference> pageLayoutDocRefOpt = getLayoutService()
+        .getLayoutPropDocRefForCurrentDoc();
     VelocityContext vcontext = ((VelocityContext) context.get("vcontext"));
     List<CSS> cssList = Collections.emptyList();
 
-    if ((pageLayoutDoc != null) && (vcontext != null)) {
+    if (pageLayoutDocRefOpt.isPresent() && (vcontext != null)) {
       List<BaseObject> baseList = new ArrayList<>();
-      baseList.addAll(addUserSkinCss(pageLayoutDoc));
-      baseList.addAll(addUserSkinCss((Document) vcontext.get("after_pagelayout_cssdoc")));
+      try {
+        XWikiDocument pageLayoutDoc = getModelAccess().getDocument(pageLayoutDocRefOpt.get());
+        baseList.addAll(addUserSkinCss(pageLayoutDoc));
+      } catch (DocumentNotExistsException dne) {
+        LOGGER.info("includeCSSAfterPageLayout pageLayoutDoc {} does not exist",
+            pageLayoutDocRefOpt.get(), dne);
+      }
       baseList.addAll(addUserSkinCss((DocumentReference) vcontext.get(
           "after_pagelayout_cssdocref")));
       cssList = includeCSS(css, "cel_css_list_pagelayout", baseList, context);
     }
     return cssList;
   }
 
-  /**
-   * CAUTION!!! needs programming rights!!
-   *
-   * @deprecated 2.81 instead use {@link addUserSkinCss(DocumentReference)}
-   */
-  @Deprecated
-  private List<BaseObject> addUserSkinCss(Document docAPI) {
-    if ((docAPI != null) && (docAPI.getDocument() != null)) {
-      return addUserSkinCss(docAPI.getDocument());
-    }
-    return Collections.emptyList();
-  }
-
   private List<BaseObject> addUserSkinCss(DocumentReference docRef) {
```

Comment:
Is this still in use, if support for skins has been dropped?

Reply:
nope